search/search-client-connect/src/main/java/io/camunda/search/connect/os/OpensearchConnector.java

```diff
@@ -0,0 +1,197 @@
+import io.camunda.search.connect.os.json.SearchRequestJacksonJsonpMapperWrapper;
+import io.camunda.search.connect.util.SecurityUtil;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.apache.hc.client5.http.auth.AuthScope;
+import org.apache.hc.client5.http.auth.UsernamePasswordCredentials;
+import org.apache.hc.client5.http.config.RequestConfig;
+import org.apache.hc.client5.http.impl.async.HttpAsyncClientBuilder;
+import org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider;
+import org.apache.hc.client5.http.impl.nio.PoolingAsyncClientConnectionManagerBuilder;
+import org.apache.hc.client5.http.ssl.ClientTlsStrategyBuilder;
+import org.apache.hc.client5.http.ssl.NoopHostnameVerifier;
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.hc.core5.util.Timeout;
+import org.opensearch.client.opensearch.OpenSearchAsyncClient;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.OpenSearchTransport;
+import org.opensearch.client.transport.aws.AwsSdk2Transport;
+import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
+import org.opensearch.client.transport.httpclient5.ApacheHttpClient5TransportBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+
+public final class OpensearchConnector {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(OpensearchConnector.class);
+
+  private final ConnectConfiguration configuration;
+  private final ObjectMapper objectMapper;
+
+  public OpensearchConnector(final ConnectConfiguration configuration) {
+    this(configuration, new JacksonConfiguration(configuration).createObjectMapper());
+  }
+
+  public OpensearchConnector(
+      final ConnectConfiguration configuration, final ObjectMapper objectMapper) {
+    this.configuration = configuration;
+    this.objectMapper = objectMapper;
+  }
+
+  public OpenSearchClient createClient() {
+    final var transport = createTransport(configuration);
+    return new OpenSearchClient(transport);
+  }
+
+  public OpenSearchAsyncClient createAsyncClient() {
+    final var transport = createTransport(configuration);
+    return new OpenSearchAsyncClient(transport);
+  }
+
+  private OpenSearchTransport createTransport(final ConnectConfiguration configuration) {
+    if (shouldCreateAWSBasedTransport()) {
+      return createAWSBasedTransport(configuration);
+    } else {
+      return createDefaultTransport(configuration);
+    }
+  }
+
+  private OpenSearchTransport createAWSBasedTransport(final ConnectConfiguration configuration) {
+    final var httpHost = getHttpHost(configuration);
+    final var region = new DefaultAwsRegionProviderChain().getRegion();
+    final var httpClient = ApacheHttpClient.builder().build();
```

Comment:
I recall I had an issue with `clearScroll` requests with `software.amazon.awssdk.http.apache.ApacheHttpClient` . I had to use `software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient` to fix it. Checkout https://github.com/opensearch-project/opensearch-java/issues/712

Reply:
@houssain-barouni, thanks for making me aware of it. I applied the change, please have another look.